### PR TITLE
[WIP] Introduce functional tests

### PR DESCRIPTION
**Pull request by pamil**
HATE THIS.

### CODE REVIEW

**Comment by pamil** [comment]
```
$ tests/Application/bin/console doc:sch:cr -vvv --env=test

ATTENTION: This operation should not be executed in a production environment.

Creating database schema...


  [Doctrine\ORM\Tools\ToolsException]
  Schema-Tool failed with Error 'An exception occurred while executing 'CREATE UNIQUE INDEX locale_foreign_key_idx ON pim_catalog_attribute_group_translation (locale, foreign_
  key)':
  SQLSTATE[HY000]: General error: 1 index locale_foreign_key_idx already exists' while executing DDL: CREATE UNIQUE INDEX locale_foreign_key_idx ON pim_catalog_attribute_group
  _translation (locale, foreign_key)
```

**Comment by pamil** [comment]
Same happens when trying to use `pim:installer:db`.

**Comment by pamil** [comment]
As for now - no tests for Akeneo side :(

File: tests/Connector/Listener/AssociationTypeSavedListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /** @var TraceableProducer */
+    private $publisher;
+
+    /** {@inheritdoc} */
+    protected function setUp(): void
+    {
+        self::bootKernel();
+
+        $this->objectManager = static::$kernel->getContainer()->get('doctrine')->getManager();
+        $this->publisher = static::$kernel->getContainer()->get('app.producer.sylius_workshop_croatia');
```
Comment (lchrusciel) [inline tests/Connector/Listener/AssociationTypeSavedListenerTest.php]: `sylius_workshop_croatia` ?

**Pull request closed without merging**